From 250d1a2668167de24570020c579fe972741f79b3 Mon Sep 17 00:00:00 2001
From: Carl Smedstad <carl.smedstad@protonmail.com>
Date: Sat, 10 Aug 2024 17:10:36 +0200
Subject: [PATCH 08/12] Do not build system_controls tables

This feature is not compatible with glibc >= 2.32 due to depending on
the header <sys/sysctl.h> which has been removed.

See: https://github.com/osquery/osquery/issues/7823
---
 osquery/tables/system/CMakeLists.txt | 4 ++--
 specs/CMakeLists.txt                 | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/osquery/tables/system/CMakeLists.txt b/osquery/tables/system/CMakeLists.txt
index c9a79c911..c03077f2b 100644
--- a/osquery/tables/system/CMakeLists.txt
+++ b/osquery/tables/system/CMakeLists.txt
@@ -43,7 +43,7 @@ function(generateOsqueryTablesSystemSystemtable)
       posix/smbios_utils.cpp
       posix/sudoers.cpp
       posix/suid_bin.cpp
-      posix/system_controls.cpp
+      # posix/system_controls.cpp
       posix/ulimit_info.cpp
     )
   endif()
@@ -83,7 +83,7 @@ function(generateOsqueryTablesSystemSystemtable)
       linux/shared_memory.cpp
       linux/smbios_tables.cpp
       linux/startup_items.cpp
-      linux/sysctl_utils.cpp
+      # linux/sysctl_utils.cpp
       linux/system_info.cpp
       linux/usb_devices.cpp
       linux/user_groups.cpp
diff --git a/specs/CMakeLists.txt b/specs/CMakeLists.txt
index 48b2e8880..4510d681f 100644
--- a/specs/CMakeLists.txt
+++ b/specs/CMakeLists.txt
@@ -245,7 +245,7 @@ function(generateNativeTables)
     "posix/socket_events.table:linux,macos"
     "posix/sudoers.table:linux,macos"
     "posix/suid_bin.table:linux,macos"
-    "posix/system_controls.table:linux,macos"
+    # "posix/system_controls.table:linux,macos"
     "posix/ulimit_info.table:linux,macos"
     "posix/usb_devices.table:linux,macos"
     "posix/user_events.table:linux,macos"
-- 
2.46.0

